extensions/mssql/src/objectExplorerNodeProvider/connection.ts

```diff
@@ -6,226 +6,93 @@
-				messages: undefined,
-				serverInfo: undefined
-			};
-		}
-		return summary;
-	}
-
-	private getConnectError(error: string | Error): string {
-		let errorMsg = utils.getErrorMessage(error);
-		if (errorMsg.indexOf('ETIMEDOUT') > -1) {
-			errorMsg = LocalizedConstants.msgTimeout;
-		} else if (errorMsg.indexOf('ENOTFOUND') > -1) {
-			errorMsg = LocalizedConstants.msgTimeout;
-		}
-		return localize('connectError', 'Connection failed with error: {0}', errorMsg);
-	}
-
-	private getEmptyServerInfo(): sqlops.ServerInfo {
-		let info: sqlops.ServerInfo = {
-			serverMajorVersion: 0,
-			serverMinorVersion: 0,
-			serverReleaseVersion: 0,
-			engineEditionId: 0,
-			serverVersion: '',
-			serverLevel: '',
-			serverEdition: '',
-			isCloud: false,
-			azureVersion: 0,
-			osVersion: '',
-			options: {}
-		};
-		return info;
-	}
-
-	public get connectionId(): string {
-		return this._connectionId;
-	}
-
-	public get host(): string {
-		if (!this._host) {
-			this.ensureHostAndPort();
-		}
-		return this._host;
-	}
+	private _port: string;
+	private _user: string;
+	private _pass: string;
 
-	/**
-	 * Sets host and port values, using any ',' or ':' delimited port in the hostname in
-	 * preference to the built in port.
-	 */
-	private ensureHostAndPort(): void {
-		this._host = this.connectionInfo.options[constants.hostPropName];
-		this._knoxPort = Connection.getKnoxPortOrDefault(this.connectionInfo);
-		// determine whether the host has either a ',' or ':' in it
-		this.setHostAndPort(',');
-		this.setHostAndPort(':');
-	}
-
-	// set port and host correctly after we've identified that a delimiter exists in the host name
-	private setHostAndPort(delimeter: string): void {
-		let originalHost = this._host;
-		let index = originalHost.indexOf(delimeter);
-		if (index > -1) {
-			this._host = originalHost.slice(0, index);
-			this._knoxPort = originalHost.slice(index + 1);
-		}
-	}
-
-	public get user(): string {
-		return this.connectionInfo.options[constants.userPropName];
-	}
-
-	public get password(): string {
-		return this.connectionInfo.options[constants.passwordPropName];
-	}
-
-	public get knoxport(): string {
-		if (!this._knoxPort) {
-			this.ensureHostAndPort();
+	constructor(connectionInfo: sqlops.connection.Connection | sqlops.IConnectionProfile) {
+		if (!this.isValid(connectionInfo)) {
+			throw new Error(localize('invalidConnectionInfo', 'Invalid ConnectionInfo is provided.'));
 		}
-		return this._knoxPort;
-	}
-
-	private static getKnoxPortOrDefault(connInfo: sqlops.ConnectionInfo): string {
-		let port = connInfo.options[constants.knoxPortPropName];
-		if (!port) {
-			port = constants.defaultKnoxPort;
-		}
-		return port;
-	}
-
-	public get connecttimeout(): number {
-		let timeoutSeconds: number = this.connectionInfo.options['connecttimeout'];
-		if (!timeoutSeconds) {
-			timeoutSeconds = constants.hadoopConnectionTimeoutSeconds;
+		if ('id' in connectionInfo) {
+			this._sqlClusterConnProfile = connectionInfo;
```

Comment:
The profile isn't needed based on looking at this code. It's only used inside this class. All you need to do is take the 'id' from connectionInfo and copy to a 'connectionId' field and you're done. Please remove this field and the public accessor given this, thanks!